diff -aurN gtkmathview-0.7.4.orig/viewer/Makefile.am gtkmathview-0.7.4/viewer/Makefile.am
--- gtkmathview-0.7.4.orig/viewer/Makefile.am	2005-08-14 21:31:13.000000000 +0200
+++ gtkmathview-0.7.4/viewer/Makefile.am	2005-10-10 16:47:31.000000000 +0200
@@ -8,9 +8,6 @@
 
 noinst_PROGRAMS = $(NULL)
 if COND_GMETADOM
-if COND_PROFILE
-noinst_PROGRAMS += mathmlviewer.static
-endif
 endif
 if COND_LIBXML2
 noinst_PROGRAMS += test_embedding
@@ -24,17 +21,6 @@
 
 noinst_HEADERS = guiGTK.h
 
-mathmlviewer_static_SOURCES = main.cc aux.cc guiGTK.c
-mathmlviewer_static_LDFLAGS = -no-install -static
-mathmlviewer_static_LDADD = \
-  $(POPT_LIBS) \
-  $(DOM_LIBS) \
-  $(GTK_LIBS) \
-  $(top_builddir)/src/widget/libgtkmathview_gmetadom.la \
-  $(T1_LIBS) \
-  $(T1_LIBS_X) \
-  $(NULL)
-
 mathmlviewer_SOURCES = main.cc aux.cc guiGTK.c
 mathmlviewer_LDADD = \
   $(POPT_LIBS) \
@@ -46,7 +32,7 @@
   $(NULL)
 
 test_embedding_SOURCES = test_embedding.c
-test_embedding_LDFLAGS = -no-install -static
+test_embedding_LDFLAGS = -no-install
 test_embedding_LDADD = \
   -lstdc++ \
   $(XML_LIBS) \
@@ -57,7 +43,7 @@
   $(NULL)
 
 test_loading_SOURCES = test_loading.c
-test_loading_LDFLASG = -no-install -static
+test_loading_LDFLASG = -no-install
 test_loading_LDADD = \
   -lstdc++ \
   $(XML_LIBS) \
@@ -68,7 +54,7 @@
   $(NULL)
 
 test_building_SOURCES = test_building.cc
-test_building_LDFLASG = -no-install -static
+test_building_LDFLASG = -no-install
 test_building_LDADD = \
   -lstdc++ \
   $(XML_LIBS) \
@@ -79,7 +65,7 @@
   $(NULL)
 
 test_rendering_SOURCES = test_rendering.cc
-test_rendering_LDFLASG = -no-install -static
+test_rendering_LDFLASG = -no-install
 test_rendering_LDADD = \
   -lstdc++ \
   $(XML_LIBS) \
@@ -92,7 +78,7 @@
   $(NULL)
 
 test_loading_reader_SOURCES = test_loading_reader.c
-test_loading_reader_LDFLASG = -no-install -static
+test_loading_reader_LDFLASG = -no-install
 test_loading_reader_LDADD = \
   -lstdc++ \
   $(XML_LIBS) \
@@ -120,4 +106,3 @@
   -I$(top_srcdir)/src/backend/common/ \
   -I$(top_srcdir)/src/backend/gtk/ \
   $(NULL)
-
