--- gtkmathview-0.6.4/configure.ac.orig	2004-11-10 13:23:56.000000000 +0100
+++ gtkmathview-0.6.4/configure.ac	2005-02-14 00:04:54.497926984 +0100
@@ -318,11 +318,7 @@
 fi
 
 dnl Checks for libraries
-AC_CHECK_LIB(iberty, getopt_long,
-	LIBS="$LIBS -liberty",
-	,
-	-liberty
-)
+AC_SEARCH_LIBS(getopt_long, iberty)
 AC_CHECK_FUNC(getopt_long, AC_DEFINE(HAVE_GETOPT_LONG,1,[Define to 1 if you have getopt_long]))
 
 AC_LANG([C])
--- gtkmathview-0.6.4/src/frontend/libxml2/Makefile.am.orig	2004-07-31 15:26:02.000000000 +0200
+++ gtkmathview-0.6.4/src/frontend/libxml2/Makefile.am	2005-02-14 23:31:04.714704448 +0100
@@ -8,6 +8,7 @@
 libmathview_frontend_libxml2_la_LIBADD = \
   $(XML_LIBS) \
   $(top_builddir)/src/frontend/common/libfrontend_common.la \
+  $(top_builddir)/src/libmathview.la \
   $(NULL)
 
 libmathview_frontend_libxml2_la_SOURCES = \
--- gtkmathview-0.6.4/src/view/Makefile.am.orig	2004-07-28 10:19:36.000000000 +0200
+++ gtkmathview-0.6.4/src/view/Makefile.am	2005-02-14 23:30:28.848156992 +0100
@@ -23,6 +23,7 @@
   $(top_builddir)/src/formatter/libformatter.la \
   $(top_builddir)/src/engine/libengine.la \
   $(top_builddir)/src/common/libcommon.la \
+  $(GTK_LIBS) \
   $(NULL)
 
 libmathview_custom_reader_la_LIBADD = \
--- gtkmathview-0.6.4/src/Makefile.am.orig	2004-06-29 16:37:00.000000000 +0200
+++ gtkmathview-0.6.4/src/Makefile.am	2005-02-14 23:31:40.597249472 +0100
@@ -10,5 +10,6 @@
 libmathview_la_LIBADD = \
   $(top_builddir)/src/formatter/libformatter.la \
   $(top_builddir)/src/engine/libengine.la \
-  $(top_builddir)/src/common/libcommon.la
+  $(top_builddir)/src/common/libcommon.la \
+  $(GTK_LIBS)
 
